### DISCUSSION TITLE
What does the brew uninstall script remove?

### DISCUSSION THREAD

**Discussion by jmtakahashi** [Homebrew/discussions#393] [Casks]
I'd like to do a clean hombrew install but I don't want my casks or the package settings deleted when re-installing.

Does anyone know if running the brew uninstall script will delete the casks that were installed by homebrew?  I know that running "brew uninstall --cask [caskname]" will delete the cask, but the uninstall script doesn't say whether it will delete all the installed casks as well.  Is there a way to retain all the installed cask packages (gui's) that were installed using hombrew when uninstalling homebrew?

In addition, does running the brew uninstall script also remove all FORMULA settings and configs that were made bye the user in the formula config files?

Is there way to retain the settings and the casks (gui's) when uninstalling and re-installing homebrew?  Thanks!

**Comment by SMillerDev** [comment]
It'll remove all installed software through homebrew, although some casks might persist.

I'm not sure why you would ever want to reinstall homebrew though. Fixing issues in `brew doctor` is almost always the better solution.

**Reply by jmtakahashi** [reply]
@SMillerDev thank you for the feedback.  Long story short, I mistakenly ran a sudo rm -rf on my /System/Volumes/Data directory while trying to remove another directory.  My data seems fine, but Homebrew was half nuked.  I had to re-run the homebrew install script and also went through the process of reinstalling and relinking formula and casks.  But I’d like to do a clean install from scratch to cleanup unused formula and un-needed files on my computer.  I was abled to re-force install all my casks so my gui programs are all ok, and their settings preserved.  But I’d like to try to retain settings on some formula that I’ll be using later, while cleaning out a few years of stuff.

**Reply by jmtakahashi** [reply]
@SMillerDev i guess its more of a question of if brew uninstall removes all my casks/guis, will my settings and preferences be preserved when I reinstall brew and reinstall those casks.

**Reply by SMillerDev** [reply]
That's really up to the individual casks since they just either remove the file, or run the upstream uninstall script.

The way I personally clean up junk that accumulated over the years is: `brew bundle dump`, remove what you don't want/need from the created `Brewfile`, `brew bundle cleanup` to validate it doesn't delete to much and finally `brew bundle cleanup --force`

**Reply by jmtakahashi** [reply]
@SMillerDev I was looking over the brew bundle docs, and was going to attempt it, but was worried about having those settings and what not preserved, as well as making sure the unnecessary files were removed at the same time.  I wasn’t sure that I could remove items from the brew file and ensure that it would delete the un-needed stuff.  What does brew bundle cleanup  do.  I understand the brew bundle dump command.

**Reply by SMillerDev** [reply]
It runs `brew remove` on anything that isn't in your `Brewfile` and isn't a dependency for one of the things in the `Brewfile`. The `brew remove` command will keep files in `HOMEBREW_PREFIX/etc` around so it's a safer option.

**Reply by jmtakahashi** [reply]
Thank @SMillerDev thats almost exactly what I need to clean my installation!  Thank you for this!
